Doc/library/collections.abc.rst

```diff
@@ -272,6 +272,11 @@ Collections Abstract Base Classes -- Detailed Descriptions
       The index() method added support for *stop* and *start*
       arguments.
 
+   .. deprecated:: 3.12
+      The :class:`ByteString` ABC has been deprecated.
```

Comment:
I would prefer to also mention PEP 688 `collections.abc.Buffer` here, once we have that available.